### ISSUE TITLE
Long duration hevc file not muxing

### ISSUE THREAD

**Issue by Learner00100** [DolbyLaboratories/dlb_mp4base#25]
Hevc file with long duration like 1hr or more not muxing and output file is 0bytes and mp4muxer.exe running where read disk activity occur for few seconds but no write disk activity.
For hevc clips like 2min duration it works fine.
Kindly solve the error, thanks.

**Comment by ShaoWeiguo** [comment]
The purpose of dlb_mp4base is to help Dolby's customers/partners to implement theirs mux tool, not use the tool directly. The executable tool provided is just for testing purpose, so it has limited quality. 
We recommend to use Bento4 which Dolby has finished the contribution, and make it support Dolby codec.

**Comment by Learner00100** [comment]
> The purpose of dlb_mp4base is to help Dolby's customers/partners to implement theirs mux tool, not use the tool directly. The executable tool provided is just for testing purpose, so it has limited quality. We recommend to use Bento4 which Dolby has finished the contribution, and make it support Dolby codec.

Thanks for information, is bento4 support dlb_mp4demux also?
Is mp4extract in bento4 is same as dlb_mp4demux?

**Comment by ShaoWeiguo** [comment]
Have a try with Mp42Hevc.

**Comment by Learner00100** [comment]
> Have a try with Mp42Hevc.

Ok thanks but it is for extract video only right, then how to extract audio like audio.ec3 in bento4 ?

**Comment by XingzhaoYun** [comment]
> > Have a try with Mp42Hevc.
> 
> Ok thanks but it is for extract video only right, then how to extract audio like audio.ec3 in bento4 ?

You can use dlb_mp4demux for ac3, ec3 and ac4 demuxing.

**Comment by ValZapod** [comment]
> read disk activity occur for few seconds but no write disk activity.

It writes stuff in RAM.

**Comment by joopie18** [comment]
@Learner00100

Have you fix this problem already?

I was using this command:

mp4muxer --dv-profile 5 -i video.hevc -i audio.ec3 --media-lang eng -o outputfile.mp4

Do you know something similar and what command should i use?

**Comment by andrei-cb** [comment]
It's working fine, just wait 5-20 minutes and it will be ready.

**Comment by joopie18** [comment]
> It's working fine, just wait 5-20 minutes and it will be ready.

Yeah i know now.  It seems like nothing happen , but i just have to wait. Thanks for answering.

**Comment by Learner00100** [comment]
> It's working fine, just wait 5-20 minutes and it will be ready.

Thank for your answer.

May I  know for mp4 demuxing which tool should I use in bento4 that function as mp4demuxer of Dolby.

**Comment by XingzhaoYun** [comment]
> 

Bento4 has mp42aac, mp42avc, mp42hevc, that's all codecs they've supported.

**Comment by XingzhaoYun** [comment]
> 
> May I know for mp4 demuxing which tool should I use in bento4 that function as mp4demuxer of Dolby.

Bento4 has mp42aac, mp42avc, mp42hevc, that's all codecs they've supported.

**Comment by Learner00100** [comment]
> > May I know for mp4 demuxing which tool should I use in bento4 that function as mp4demuxer of Dolby.
> 
> Bento4 has mp42aac, mp42avc, mp42hevc, that's all codecs they've supported.

But for ac3,ac4 what to choose?
Mp42aac only for aac right.
I need for ac3,ac4. So kindly tell what to choose.

**Comment by XingzhaoYun** [comment]
> > > May I know for mp4 demuxing which tool should I use in bento4 that function as mp4demuxer of Dolby.
> > 
> > 
> > Bento4 has mp42aac, mp42avc, mp42hevc, that's all codecs they've supported.
> 
> But for ac3,ac4 what to choose? Mp42aac only for aac right. I need for ac3,ac4. So kindly tell what to choose.
Dolby mp4demuxer might be a choice https://github.com/DolbyLaboratories/dlb_mp4demux.
Mp4box https://github.com/gpac/gpac also supports ac3/eac3 according to their page.
